pkg/vm/engine/disttae/txn_table.go

```diff
@@ -1805,3 +1805,14 @@ func (tbl *txnTable) updateLogtail(ctx context.Context) (err error) {
 	tbl.logtailUpdated = true
 	return nil
 }
+
```

Comment:
This function is tricky and need lots of comment.

What it does, why do we need it.   Error condition, what does mean.    Cost, running time, O(n) or O(n^2) and what is n here, etc.